rust/lancedb/src/error.rs

```diff
@@ -28,6 +28,8 @@ pub enum Error {
 
     #[snafu(display("Table '{name}' already exists"))]
     TableAlreadyExists { name: String },
+    #[snafu(display("Namespace error: {message}"))]
+    Namespace { message: String },
```

Comment:
Is this useful?  Couldn't the user infer that the error has something to do with namespaces from the message itself?  Or do you envision a user may someday treat namespace errors differently than something like InvalidInput?